app/models/admin_only/app_configuration.rb

```diff
@@ -1,8 +1,41 @@
 module AdminOnly
 
+  #--------------------------
+  #
+  # @class AppConfiguration
+  #
+  # @desc Responsibility: Aggregates discrete data items that are used to configure
+  #  various aspects of the system (app).
+  #  "configure" means settings that are more about customizing the application;
+  # settings that can be changed after the application is running.
+  #
+  #  This is a Singleton. (Only 1 AppConfiguration instance exists/is needed.)
+  #  This is enforced in the following ways:
+  #    1. There is an index defined on the '' column in the table, and
+  #       this index
+  #       If the database detects an attempt to create more than 1 record
+  #       (e.g. more than 1 index entry), it will throw a
+  #       ActiveRecord::RecordNotUnique exception.
+  #       See the comments for validates_uniqueness_of {ActiveRecord::Validations::UniquenessValidator::ClassMethods#validates_uniqueness_of}.
+  #       Specifically, read the section "Concurrency and integrity"
+  #    2. validates_inclusion_of :singleton_guard, in: [0]
+  #       This ensures that the value of :singleton_guard can only be 0
+  #       Combined with the unique index above, this ensures that there can
+  #       only be 1 row in the table, and the singleton_guard value is always 0.
+  #
+  #
+  # @file admin_only/app_configuration.rb
+  #
+  #--------------------------
   class AppConfiguration < ApplicationRecord
-    # Aggregates discrete data items that are used to configure
-    # various aspects of the system (app).
+
+
+    # The "singleton_guard" column is a unique column which must always be set to '0'
+    # This ensures that only one AppSettings row is created
+    validates_inclusion_of :singleton_guard, in: [0]
```

Comment:
This is clever!

Reply:
https://stackoverflow.com/questions/399447/how-to-implement-a-singleton-model

The key is ensuring the single entry in the db (of course), so I searched for that vs. just doing a standard Singleton.